common/djangoapps/student/helpers.py

```diff
@@ -268,16 +268,31 @@ def get_next_url_for_login_page(request):
     /account/finish_auth/ view following login, which will take care of auto-enrollment in
     the specified course.
 
-    Otherwise, we go to the ?next= query param or to the dashboard if nothing else is
-    specified.
+    Otherwise, we go to the ?next= query param or the configured custom
+    redirection url (the default behaviour is to go to /dashboard).
 
     If THIRD_PARTY_AUTH_HINT is set, then `tpa_hint=<hint>` is added as a query parameter.
     """
     redirect_to = _get_redirect_to(request)
     if not redirect_to:
         try:
-            redirect_to = reverse('dashboard')
+            login_redirect_url = configuration_helpers.get_value('DEFAULT_REDIRECT_AFTER_LOGIN')
```

Comment:
It seems like this could be a lot simpler if we set `dashboard` as the default value for when we don't get a valid 'DEFAULT_REDIRECT_AFTER_LOGIN'.  If we do that, we should be able to remove the nested try/catch.  And then just have one warning to indicate that the we failed to go to the `login_redirect_url`.

Reply:
@feanil I think that the main issue here is that this endpoint exists for both LMS and Studio, so the outer try-except block actually reverses to a page that exists in Studio (but not on LMS).

It would be better if there was a way to detect which environment is the request from (LMS or Studio), but I haven't found any when I implemented this change.

Reply:
It would be nice to check which environment is the request from (using an utility function) to avoid nested try-except.
I haven't analyzed the structure of the request or any helper methods to see if it's cleanly possible to do that easily.

Do you have any suggestions?

Reply:
I see this pattern used in many places to check if it's running in the LMS or 'cms': `settings.ROOT_URLCONF == 'lms.urls'`

Reply:
@xitij2000 @feanil I've changed the behavior to use  `ROOT_URLCONF`, fixed tests and quality issues and squashed the commits. This is ready for review.